simpeg/electromagnetics/time_domain/sources.py

```diff
@@ -199,31 +199,82 @@ class RampOffWaveform(BaseWaveform):
 
     """
 
-    def __init__(self, off_time=0.0, **kwargs):
-        super().__init__(off_time=off_time, has_initial_fields=True, **kwargs)
+    def __init__(self, ramp_end=0.0, ramp_start=0.0, **kwargs):
+        if (off_time := kwargs.pop("off_time", None)) is not None:
+            ramp_end = off_time
```

Comment:
There's something here that it's making my brain itch. If a user passes `RampOffWavefor(ramp_start=0.0, ramp_end=1.0, off_time=1.2)`, then the `ramp_end` will get assigned to `1.2` without any notice. Should we deprecate `off_time` directly? Also, we should update the docstrings and explain what `ramp_end` and `ram_start` are.

Reply:
Yeah, I was originally going for backward compatibility and keyword arg support, but I think the cleanest is to just do it with variable length positional args to allow both:
* `RampOffWaveform(0.1)` to have an implicit start time of 0 (also backwards compatible for positional argument passing).
* `RampOffWaveform(0.01, 0.1)` for specifying both start and end times
* `RampOffWaveform(off_time=0.1)` same as first, but issues a deprecated warning.
* `RampOffWaveform(0.01, off_time=0.1)` raises a `TypeError`
* `RampOffWaveform()` raises a `TypeError` (as well as when calling with too many positional arguments.)